db/migrate/20210512034228_add_partial_index_to_lesson_id_user_id_project_submissions.rb

```diff
@@ -0,0 +1,5 @@
+class AddPartialIndexToLessonIdUserIdProjectSubmissions < ActiveRecord::Migration[6.1]
+  def change
+    add_index :project_submissions, %i[user_id lesson_id], unique: true, where: 'discarded_at is null'
```

Comment:
Using a partial index is a super slick way of solving the one submission per project problem 👏

Reply:
:smile:  I was happy when I found this! I googled a bunch and didn't find much, and then there it was in the Rails docs haha.